ide/editor.completion/src/org/netbeans/spi/editor/completion/CompletionCollector.java

```diff
@@ -0,0 +1,565 @@
+            this.insertTextFormat = insertTextFormat;
+            this.textEdit = textEdit;
+            this.additionalTextEdits = additionalTextEdits;
+            this.commitCharacters = commitCharacters;
+            this.command = command;
+            this.data = data;
+        }
+
+        /**
+	 * The label of this completion. By default also the text that is inserted
+         * when selecting this completion.
+	 */
+        @NonNull
+        public String getLabel() {
+            return label;
+        }
+
+        /**
+	 * The kind of this completion.
+	 */
+        @CheckForNull
+        public Kind getKind() {
+            return kind;
+        }
+
+        /**
+         * Tags for this completion.
+         */
+        @CheckForNull
+        public List<Tag> getTags() {
+            return tags;
+        }
+
+        /**
+	 * A human-readable string with additional information
+	 * about this completion, like type or symbol information.
+	 */
+        @CheckForNull
+        public String getDetail() {
+            return detail;
+        }
+
+        /**
+	 * A human-readable string that represents a doc-comment.
+	 */
+        @CheckForNull
+        public String getDocumentation() {
+            return documentation;
+        }
+
+        /**
+	 * Select this completion when showing.
+	 */
+        public boolean isPreselect() {
+            return preselect;
+        }
+
+        /**
+	 * A string that should be used when comparing this completion with other
+         * completions. When {@code null} the label is used as the sort text.
+	 */
+        @CheckForNull
+        public String getSortText() {
+            return sortText;
+        }
+
+        /**
+	 * A string that should be used when filtering a set of completions.
+         * When {@code null} the label is used as the filter.
+	 */
+        @CheckForNull
+        public String getFilterText() {
+            return filterText;
+        }
+
+        /**
+	 * A string that should be inserted into a document when selecting
+	 * this completion. When {@code null} the label is used as the insert text.
+         */
+        @CheckForNull
+	public String getInsertText() {
+            return insertText;
+        }
+
+        /**
+	 * The format of the insert text. The format applies to both the
+	 * {@code insertText} property and the {@code newText} property of a provided
+	 * {@code textEdit}. If omitted defaults to {@link TextFormat#PlainText}.
+	 */
+        @CheckForNull
+        public TextFormat getInsertTextFormat() {
+            return insertTextFormat;
+        }
+
+        /**
+	 * An edit which is applied to a document when selecting this completion.
+	 * When an edit is provided the value of {@code insertText} is ignored.
+	 * The range of the edit must be a single line range and it must
+	 * contain the position at which completion has been requested.
+	 */
+        @CheckForNull
+        public TextEdit getTextEdit() {
```

Comment:
These methods do not exit by mistake in LSP. Most completion items provide just a simple `insertText`, however some completion items may need a more complex handling - replace some existing text with something different (e.g. `OverrideMethodItem` in Java which needs to remove already written name prefix and replace it with the whole method declaration usually starting with modifiers and return type). Those items provide  `textEdit` instead of `insertText`.